internal/controller/vrg_status_pvcs.go

```diff
@@ -8,17 +8,27 @@ import (
 	ramen "github.com/ramendr/ramen/api/v1alpha1"
 )
 
-// findProtectedPVC returns the &VRG.Status.ProtectedPVC[x] for the given pvcName
-func (v *VRGInstance) findProtectedPVC(pvcNamespaceName, pvcName string) *ramen.ProtectedPVC {
-	return FindProtectedPVC(v.instance, pvcNamespaceName, pvcName)
-}
-
 func FindProtectedPVC(vrg *ramen.VolumeReplicationGroup, pvcNamespaceName, pvcName string) *ramen.ProtectedPVC {
 	protectedPvc, _ := FindProtectedPvcAndIndex(vrg, pvcNamespaceName, pvcName)
 
 	return protectedPvc
 }
 
+func (v *VRGInstance) getOrCreateProtectedPVC(pvcNamespace, pvcName string, srcPVC *ramen.ProtectedPVC,
+) *ramen.ProtectedPVC {
+	protectedPVC, _ := FindProtectedPvcAndIndex(v.instance, pvcNamespace, pvcName)
+	if protectedPVC == nil {
+		protectedPVC = &ramen.ProtectedPVC{Namespace: pvcNamespace, Name: pvcName}
+		if srcPVC != nil {
+			srcPVC.DeepCopyInto(protectedPVC)
+		}
+
+		v.instance.Status.ProtectedPVCs = append(v.instance.Status.ProtectedPVCs, *protectedPVC)
+	}
+
+	return protectedPVC
```

Comment:
Note that this is *not* the same object added to the slice. Change to this object will be missing from the protectedPVC int the slice.